csharp/src/Ice/OutputStream.cs

```diff
@@ -480,6 +480,200 @@ public void WriteVarULong(ulong v)
             }
         }
 
+        /// <summary>Writes a dictionary to the stream. The dictionary's value type is reference type.</summary>
+        /// <param name="v">The dictionary to write.</param>
+        /// <param name="withBitSequence">When true, encodes entries with a null value using a bit sequence; otherwise,
+        /// false.</param>
+        /// <param name="keyWriter">The delegate that writes each key to the stream.</param>
+        /// <param name="valueWriter">The delegate that writes each non-null value to the stream.</param>
+        public void WriteDictionary<TKey, TValue>(
+            IReadOnlyDictionary<TKey, TValue?> v,
+            bool withBitSequence,
+            OutputStreamWriter<TKey> keyWriter,
+            OutputStreamWriter<TValue> valueWriter)
+            where TKey : notnull
+            where TValue : class
+        {
+            if (withBitSequence)
+            {
+                int count = v.Count();
+                WriteSize(count);
+                BitSequence bitSequence = WriteBitSequence(count);
+                int index = 0;
+                foreach ((TKey key, TValue? value) in v)
+                {
+                    keyWriter(this, key);
+                    if (value is TValue actualValue)
+                    {
+                        valueWriter(this, actualValue);
+                    }
```

Comment:
I think for reference types we can just use `value` after checking for null
```suggestion
                    if (value != null)
                    {
                        valueWriter(this,value);
                    }
```

Reply:
Fixed as suggested.